File: Python/programmers/etc/level2/연속_부분_수열_합의_개수.py
def solution(elements):
    n = len(elements)
    answer = set()

    for i in range(n):
        tmp = elements[i]
        answer.add(tmp)
        for j in range(i+1, i+n):
            tmp += elements[j%n]
            answer.add(tmp)
    
    return len(answer)


elements = [7,9,1,1,4]
print(solution(elements)) # 18

File: Java/programmers/etc/level2/연속부분수열합의개수.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 연속부분수열합의개수 {
    public int solution(int[] elements) {
        int n = elements.length;
        HashSet<Integer> answer = new HashSet<>();
        
        for (int i=0; i<n; i++) {
            int tmp = elements[i];
            answer.add(tmp);
            for (int j=i+1; j<i+n; j++) {
                tmp += elements[j%n];
                answer.add(tmp);
            }
        }
        
        return answer.size();
    }
}
